.github/workflows/puppet-test.yml

```diff
@@ -0,0 +1,85 @@
+name: puppet-test
+
+# Only run installer tests, Rake spec, lint for main branch, tags, or if the branch has relevant changes
+on: 
+  push:
+    tags:
+      - '*'
+    paths:
```

Comment:
Let's also include this workflow file itself.

Reply:
Changes made.